packages/core/contracts/insured-bridge/BridgePool.sol

```diff
@@ -218,6 +219,135 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
+        require(
+            depositData.slowRelayFeePct < 0.25e18 &&
+                depositData.instantRelayFeePct < 0.25e18 &&
+                realizedLpFeePct < 0.5e18
+        );
+
+        // Check if there is a pending relay for this deposit.
+        bytes32 depositHash = _getDepositHash(depositData);
+
+        // Note: A disputed relay deletes the stored relay hash and enables this require statement to pass.
+        require(relays[depositHash] == bytes32(0), "Pending relay exists");
+
+        uint256 proposerBond = _getProposerBond(depositData.amount);
+        uint256 finalFee = store.computeFinalFee(address(l1Token)).rawValue;
+
+        // Save hash of new relay attempt parameters.
+        RelayData memory relayData =
+            RelayData({
+                relayState: RelayState.Pending,
+                slowRelayer: msg.sender, // Note: Increment numberOfRelays at the same time as setting relayId to its current value.
+                relayId: numberOfRelays++,
+                realizedLpFeePct: realizedLpFeePct,
+                priceRequestTime: priceRequestTime,
+                proposerBond: proposerBond,
+                finalFee: finalFee
+            });
+        bytes32 relayHash = _getRelayHash(depositData, relayData);
+        relays[depositHash] = _getRelayDataHash(relayData);
+
+        // Sanity check that pool has enough balance to cover relay amount + proposer reward. Reward amount will be
+        // paid on settlement after the OptimisticOracle price request has passed the challenge period.
+        require(
+            l1Token.balanceOf(address(this)) >= depositData.amount + proposerBond &&
+                liquidReserves >= depositData.amount + proposerBond,
+            "Insufficient pool balance"
+        );
+
+        // Compute total proposal bond and pull from caller so that the OptimisticOracle can pull it from here.
+        uint256 totalBond = proposerBond + finalFee;
+        l1Token.safeTransferFrom(msg.sender, address(this), depositData.amount + totalBond);
+        bonds += totalBond;
+
+        // Request a price for the relay identifier and propose "true" optimistically. This method will pull the
+        // (proposer reward + proposer bond + final fee) from the caller. We need to set a new price request timestamp
+        // instead of default setting to equal to the `depositTimestamp`, which is dependent on the L2 VM on which the
+        // DepositContract is deployed. Imagine if the timestamps on the L2 have an offset that are always "in the
+        // future" relative to L1 blocks, then the OptimisticOracle would always reject requests.
+        // _requestAndProposeOraclePriceRelay(proposerBond, priceRequestTime, ancillaryData);
+
+        pendingReserves += depositData.amount; // Book off maximum liquidity used by this relay in the pending reserves.
+
+        bytes32 instantRelayHash = _getInstantRelayHash(depositHash, relayData);
+        require(
```

Comment:
Surly this require is redundant for relay and speed up? You have already validated this by the fact that there is no relay data, no? Alternatively, If you think you need this check, you should put it higher up if possible to revert sooner

Reply:
I think this check is necessary. You could have a previous identical relay that was falsely disputed that was already sped up.

Will move up (@nicholaspai made a similar comment in the other PR)!